R/counting_process.R

```diff
@@ -97,50 +97,53 @@ counting_process <- function(x, arm) {
     stop("counting_process: event indicator must be 0 (censoring) or 1 (event).")
   }
 
+  # initilize the number of subjects at risk
   ans <- as.data.table(x)
   ans <- ans[order(tte, decreasing = TRUE), ]
   ans[, one := 1]
   ans[, `:=`(
-    n_risk_tol = cumsum(one),
+    n_risk_total = cumsum(one),
     n_risk_trt = cumsum(treatment == arm)
   ), by = "stratum"]
 
   # Handling ties using Breslow's method
   if (uniqueN(ans[, .(stratum, tte)]) < nrow(ans)) { # ties
     ans[, mtte := -tte]
     ans <- ans[, .(
-      events = sum(event),
-      n_event_tol = sum((treatment == arm) * event),
+      event_total = sum(event),
+      event_trt = sum((treatment == arm) * event),
       tte = tte[1],
-      n_risk_tol = max(n_risk_tol),
+      n_risk_total = max(n_risk_total),
       n_risk_trt = max(n_risk_trt)
     ), by = c("stratum", "mtte")]
     ans[, mtte := NULL]
   } else { # no ties
     ans <- ans[, .(
       stratum,
-      events = event,
-      n_event_tol = (treatment == arm) * event,
+      event_total = event,
+      event_trt = (treatment == arm) * event,
       tte,
-      n_risk_tol,
+      n_risk_total,
       n_risk_trt
     )]
   }
 
   # Keep calculation for observed time with at least one event,
   # at least one subject is at risk in both treatment group and control group.
-  ans <- ans[events > 0 & n_risk_tol - n_risk_trt > 0 & n_risk_trt > 0, ]
-  ans[, s := 1 - events / n_risk_tol]
+  ans <- ans[event_total > 0 & n_risk_total - n_risk_trt > 0 & n_risk_trt > 0, ]
+  ans[, s := 1 - event_total / n_risk_total]
   ans <- ans[order(stratum, tte), ]
   # Left continuous Kaplan-Meier Estimator
   ans[, s := c(1, cumprod(s)[-length(s)]), by = "stratum"]
   # Observed events minus Expected events in treatment group
-  ans[, o_minus_e := n_event_tol - n_risk_trt / n_risk_tol * events]
+  ans[, o_minus_e := event_trt - n_risk_trt / n_risk_total * event_total]
   # Variance of o_minus_e
-  ans[, var_o_minus_e := (n_risk_tol - n_risk_trt) *
-    n_risk_trt * events * (n_risk_tol - events) /
-    n_risk_tol^2 / (n_risk_tol - 1)]
+  ans[, var_o_minus_e := (n_risk_total - n_risk_trt) *
+    n_risk_trt * event_total * (n_risk_total - event_total) /
+    n_risk_total^2 / (n_risk_total - 1)]
 
   setDF(ans)
+  class(ans) <- c("counting_process", class(ans))
```

Comment:
What is the purpose of adding the class `"counting_process"` to the returned object? There are no S3 methods defined for this class. Until S3 methods are defined, I think we should keep things simple and not create unused classes.

Reply:
Thanks for sharing your insights, @jdblischak! It is a request from Keaven. We will develop some S3 method for `counting_process` later in the future.

Here are some further information.
- If we have a counting_process object, then we can directly get the test statistics.
- If we have original TTE data, we should transfer it to a counting_process object and then to test.

Reply:
> We will develop some S3 method for `counting_process` later in the future.

My argument is that adding the class to the output object should be added in the same future PR that adds the methods. This PR only renames the output columns, and would be cleaner if it was limited to only that change.

Reply:
Summary from Teams discussion: `counting_process` class is needed now even though no S3 methods yet exist